OpenRA.Mods.Common/Widgets/Logic/ServerListLogic.cs

```diff
@@ -762,14 +762,19 @@ static int ListOrder(GameServer g)
 
 							if (game.Clients.Length > 0)
 							{
-								var displayClients = game.Clients.Select(c => c.Name);
-								if (game.Clients.Length > 10)
-									displayClients = displayClients
-										.Take(9)
-										.Append(FluentProvider.GetMessage(OtherPlayers, "players", game.Clients.Length - 9));
-
-								var tooltip = displayClients.JoinWith("\n");
-								players.GetTooltipText = () => tooltip;
+								var preview = modData.MapCache[game.Map];
+								var tooltip = new CachedTransform<MapStatus, string>(s =>
+								{
+									var displayClients = game.Clients.Select(c => c.IsBot ? preview.GetMessage(c.Name) : c.Name);
```

Comment:
We should be returning a placeholder "AI Player"(?) rather than the bot id if we can't yet translate it.

Reply:
that's a good idea

Reply:
I'd actually be fine with hardcoding "AI Player" for *all* bots outside of the lobby screen.

Reply:
What about a case where a map simply doesn't use fluent? We need extra handling

Reply:
Show "AI Player" for them too.

Reply:
Now using Unnamed Bot